frontend/src/metabase/visualizations/components/TableInteractive/TableInteractive.jsx

```diff
@@ -482,10 +486,20 @@ class TableInteractive extends Component {
   }
 
   pkClick = rowIndex => {
-    const objectId = this.state.IDColumn
-      ? this.props.data.rows[rowIndex][this.state.IDColumnIndex]
-      : rowIndex;
-    return e => this.props.onZoomRow(objectId);
+    let objectId;
+
+    if (this.state.IDColumn) {
+      objectId = this.props.data.rows[rowIndex][this.state.IDColumnIndex];
+    } else {
+      const map = this.props.PKRowIndexMap;
+
+      objectId =
+        Object.keys(map).find(key => map[key] === rowIndex) ?? rowIndex;
+    }
+
+    return () => {
+      this.props.onZoomRow(objectId);
+    };
```

Comment:
I don't know if there's a problem, but I'd feel safer if all `this.props` and `this.state` references happened inside the returned callback (to avoid referencing stale props when callback is executed).

```suggestion
    return () => {
      let objectId;

      if (this.state.IDColumn) {
        objectId = this.props.data.rows[rowIndex][this.state.IDColumnIndex];
      } else {
        const map = this.props.PKRowIndexMap;

        objectId =
          Object.keys(map).find(key => map[key] === rowIndex) ?? rowIndex;
      }

      this.props.onZoomRow(objectId);
    };
```

Reply:
well, this function is re-calculated every time you hover over a row

`hoverDetailEl.onclick = this.pkClick(rowIndex)` 

or press "Enter" - `this.pkClick(hoveredRowIndex)(event);` 

I'd not worry about stale props

Reply:
changed it in fa0289f5644f3b912a901a4bdafc58f65468e797